File: Go/Linked List/1669_Merge In Between Linked Lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
	cur := list1
	temp := &ListNode{}
	count := 0
	for cur != nil {
		if count == a-1 {
			temp = cur.Next
			cur.Next = list2
			break
		}
		cur = cur.Next
		count += 1
	}
	count += 1
	temp1 := cur
	for temp1.Next != nil {
		temp1 = temp1.Next
	}
	for temp != nil {
		if count == b+1 {
			temp1.Next = temp
		}
		count += 1
		temp = temp.Next
	}
	return list1
}

File: C++/Linked List/1669_MergeInBetweenLinkedLists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *mergeInBetween(ListNode *list1, int a, int b, ListNode *list2)
    {
        int count = 0;
        ListNode *temp = list1;
        ListNode *start;
        while (temp != NULL && count <= b)
        {
            if (count == a - 1)
                start = temp;
            temp = temp->next;
            count++;
        }
        start->next = list2;
        while (list2->next != NULL)
            list2 = list2->next;
        list2->next = temp;
        return list1;
    }
};
